org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -82,6 +86,9 @@
 	@Inject 
 	private OperationCanceledManager operationCanceledManager;
 	
+	@Inject
+	private BuilderPreferenceAccess preferenceAccess;
```

Comment:
@szarnekow isnt this a language specific thing?

Reply:
1) No implementation for java.lang.String annotated with @com.google.inject.name.Named(value=languageName) was bound.
  while locating java.lang.String annotated with @com.google.inject.name.Named(value=languageName)
    for parameter 0 at org.eclipse.xtext.ui.editor.preferences.PreferenceStoreAccessImpl.setLanguageNameAsQualifier(Unknown Source)
  at org.eclipse.xtext.ui.editor.preferences.PreferenceStoreAccessImpl.setLanguageNameAsQualifier(Unknown Source)
  while locating org.eclipse.xtext.ui.editor.preferences.IPreferenceStoreAccess
    for parameter 0 at org.eclipse.xtext.builder.preferences.BuilderPreferenceAccess.setPreferenceStoreAccess(Unknown Source)
  at org.eclipse.xtext.builder.preferences.BuilderPreferenceAccess.setPreferenceStoreAccess(Unknown Source)

Reply:
=> we should not use preference access for this?

Reply:
True. This should use the standard APIs to access the preference.

Reply:
Will remove this from BuilderPreferenceAccess and use IPreferenceStoreAccess in the XtextBuilder instead.